### DISCUSSION TITLE
email Tracking is not working with NGINX Proxy Manager

### DISCUSSION THREAD

**Discussion by drtech981** [postalserver/postal#1693] [Help with using Postal] [answered]
I am trying to set up click based email tracking using NGINx Proxy Manager as the frontend, however it has been failing miserably.

DNS records : 
track.services.drtech.cc A 103.121.118.181
click.services.drtech.cc CNAME track.services.ahmads.cc

NGINX Proxy Manager Config : 
![image](https://user-images.githubusercontent.com/79300765/140797169-7a77c4a7-c76c-4ad1-8afd-fb9aaa83c3cd.png)

Headers : 
![image](https://user-images.githubusercontent.com/79300765/140797281-bfe66b11-2a43-4cac-9f34-c5f38e9f95ca.png)

Error on tracked links : 

![image](https://user-images.githubusercontent.com/79300765/140797351-ff819b34-0a09-4464-8278-1ab0cee149a2.png)

What am I doing wrong ?

**Comment by willpower232** [comment]
I think that final screenshot means the header you added is not getting through to the application. I don't know enough about Nginx Proxy Manager to know why it wouldn't work.

**Comment by drtech981** [comment, marked as answer]
Solved : 

`add_header X-Postal-Track-Host 1;`
adds a header downstream the proxy, that is, on the page returned by the reverse proxy to the browser.

You need to add : 
`proxy_set_header X-Postal-Track-Host 1;`
adds the required header upstream, that is, to the request being proxied to the upstream/backed postal server.

![image](https://user-images.githubusercontent.com/79300765/142491544-18f1cad2-7a03-4911-9506-697b3f1fa7c7.png)

This is all you need to add to get it working.
